article/_posts/2019-11-21-stealml.md

```diff
@@ -0,0 +1,87 @@
+
+
+![](/article/images/stealml/tab4.jpeg)
+
+## Online Model Extraction Attacks
+
+The authors then tried to extract models setup online on BigML and Amazon AWS. BigML uses decision tree which will not be adressed in this review. Amazon, however, allows users to upload their data and then Amazon will perform feature extraction and then train a model on these features.
+
+The problem now is to find what features were extracted and which model was used. However, AWS does support "partial input", where you can send an image for prediction which does not have some of the extracted features and still get a result. Without going into details, the authors were able to reverse-engineer the features and still get a copy of the model with 100% matches.
+
+![](/article/images/stealml/tab7.jpeg)
+
+In all the attacks, the authors were able to learn about the model's class, the enabled feature extractors and hyperparameters through the ML service AWS's documentation. In a real world scenario, an attacker could narrow down its hyperparameter search through similar techniques and cross-validation.
+
+## Extraction Given Class Labels Only
+
+Without going into details, the authors wanted to see if they could extract model through only class predictions. The general strategy is called retraining, where the authors query the models with data points that are close to $$\hat{f}$$'s decision boundary.
+
+![](/article/images/stealml/fig4.jpeg)
+![](/article/images/stealml/fig5.jpeg)
+![](/article/images/stealml/fig6.jpeg)
+
+where $$\alpha$$ is the budget, or multiplier of $$(d+1)$$ data points used.
+
+##  Prevention from such attacks
+
+The authors suggest some countermeasures to such attacks, such as rounding off the prediction's accuracy. The impacts of rounding the accuracy (or only providing class labels) is reported below
+
+![](/article/images/stealml/fig7.jpeg)
+
+They also mention using ensemble methods, as recovering $$n$$ times model parameters is much more costly and only one set of model parameters.
+
+# Conclusions
+> [The authors] demonstrated how the flexible prediction APIs exposed by current ML-as-a-service providers enable new model extraction attacks that could subvert model monetization, violate training-data privacy, and facilitate model evasion.
+
+# Remarks
+
+- All the attacks were done on models setup by the authors themselves so as not to potentially harm anyone.
```

Comment:
*setup* is the noun; *set up* is the verb if I'm not mistaken

Reply:
fixed in a5fac43